sdks/python/apache_beam/io/iobase.py

```diff
@@ -1165,3 +1171,56 @@ def check_done(self):
       ~exceptions.ValueError: if there is still any unclaimed work remaining.
     """
     raise NotImplementedError
+
+
+class RestrictionProgress(object):
+  """Used to record the progress of a restriction.
+
+  Experimental; no backwards-compatibility guarantees.
+  """
+  def __init__(self, **kwargs):
+    # Only accept keyword arguments.
+    self._fraction = kwargs.pop('fraction', None)
```

Comment:
Should we have this be a cumulative count of number of elements seen so far, or a fraction?